src/sdk/pynni/nni/metis_tuner/metis_tuner.py

```diff
@@ -0,0 +1,434 @@
+
+logger = logging.getLogger("Metis_Tuner_AutoML")
+
+@unique
+class OptimizeMode(Enum):
+    '''
+    Optimize Mode class
+    '''
+    Minimize = 'minimize'
+    Maximize = 'maximize'
+
+
+NONE_TYPE = ''
+CONSTRAINT_LOWERBOUND = None
+CONSTRAINT_UPPERBOUND = None
+CONSTRAINT_PARAMS_IDX = []
+
+
+class MetisTuner(Tuner):
+    '''
+    Metis Tuner
+    '''
+
+    def __init__(self, optimize_mode="maximize", no_resampling=True, no_candidates=True,
+                 selection_num_starting_points=10, cold_start_num=10):
+        '''
+        optimize_mode: is a string that including two mode "maximize" and "minimize"
+        key_order: is a list that contains all name of parameters, same index as samples_x
+        no_resampling: True or False
+        no_candidates: True or False
+        '''
+        self.samples_x = []
+        self.samples_y = []
+        self.samples_y_aggregation = []
+        self.space = None
+        self.no_resampling = no_resampling
+        self.no_candidates = no_candidates
+        self.optimize_mode = optimize_mode
+        self.key_order = []
+        self.cold_start_num = cold_start_num
+        self.selection_num_starting_points = selection_num_starting_points
+        self.minimize_constraints_fun = None
+        self.minimize_starting_points = None
+
+
+    def update_search_space(self, search_space):
+        '''
+        Update the self.x_bounds and self.x_types by the search_space.json
+        '''
+        self.x_bounds = [[] for i in range(len(search_space))]
+        self.x_types = [NONE_TYPE for i in range(len(search_space))]
+
+        for key in search_space:
+            self.key_order.append(key)
+
+        key_type = {}
+        if isinstance(search_space, dict):
+            for key in search_space:
+                key_type = search_space[key]['_type']
+                key_range = search_space[key]['_value']
+                try:
+                    idx = self.key_order.index(key)
+                except Exception as ex:
+                    logger.exception(ex)
+                    raise RuntimeError("The format search space contains \
+                                        some key that didn't define in key_order.")
+
+                if key_type == 'quniform':
+                    if key_range[2] == 1:
+                        self.x_bounds[idx] = [key_range[0], key_range[1]]
+                        self.x_types[idx] = 'range_int'
+                    else:
+                        bounds = []
+                        for value in range(key_range[0], key_range[1], key_range[2]):
+                            bounds.append(value)
+                        self.x_bounds[idx] = bounds
+                        self.x_types[idx] = 'discrete_int'        
+                elif key_type == 'randint':
+                    self.x_bounds[idx] = [0, key_range[0]]
+                    self.x_types[idx] = 'range_int'
+                elif key_type == 'uniform':
```

Comment:
you support `uniform`? not mentioned in README?

Reply:
thx. add to README